src/chatClient.cpp

```diff
@@ -3789,6 +3850,88 @@ bool Client::isCallInProgress(karere::Id chatid) const
 }
 
+void Client::updateAliases(Buffer *data)
+{
+    // Save the aliases from cache attr in a tlv container
+    const std::string container(data->buf(), data->size());
+    ::mega::TLVstore *tlvRecords = ::mega::TLVstore::containerToTLVrecords(&container);
+    std::vector<std::string> *keys = tlvRecords->getKeys();
+    AliasesMap aliasesUpdated;
+
+    // Create a new map <uhBin, aliasB64> for the aliases that has been updated
+    for (int i=0; i < keys->size(); i++)
+    {
+        std::string key(keys->at(i));
+        Id handleBin(key.data());
+        if (handleBin.isValid() && key.size())
+        {
+            if (mAliasesMap[handleBin] != tlvRecords->get(key))
+            {
+                aliasesUpdated[handleBin] = tlvRecords->get(key);
+                mAliasesMap[handleBin] = tlvRecords->get(key);
+            }
+        }
+    }
+    delete tlvRecords;
+    delete keys;
+
+    // Iterate through all chatrooms and update the aliases contained in aliasesUpdated
+    ChatRoomList::iterator it;
+    for (it = chats->begin(); it != chats->end(); it++)
+    {
+        ChatRoom *chatroom = it->second;
+        AliasesMap::iterator it;
+        for (it = aliasesUpdated.begin(); it != aliasesUpdated.end(); it++)
+        {
+            uint64_t userHandle = it->first;
+            if (chatroom->isGroup())
+            {
+                // If chatroom is a group chatroom and there's at least a chat member included
+                // in aliasesUpdated map we need to re-generate the default title
+                // if there's no custom title
+                GroupChatRoom *room = static_cast<GroupChatRoom *>(chatroom);
+                if (room->hasTitle())
+                {
+                    continue;
+                }
+
+                auto it = room->peers().find(userHandle);
+                if (it != room->peers().end())
+                {
+                    room->makeTitleFromMemberNames();
+                    break;
+                }
+            }
+            else
+            {
+                PeerChatRoom *room = static_cast<PeerChatRoom *>(chatroom);
+                if (userHandle == room->peer())
+                {
+                    std::string aliasB64 = it->second;
+                    Buffer buf(aliasB64.size());
+                    size_t decLen = base64urldecode(aliasB64.c_str(), aliasB64.size(), buf.buf(), buf.bufSize());
+                    room->updateChatRoomTitle(std::string(buf.buf(), decLen));
+                }
+            }
+        }
+    }
+}
+
+std::string Client::getUserAlias(uint64_t userId)
+{
+    std::string aliasBin;
+    std::map<uint64_t, std::string>::iterator it;
+    it = mAliasesMap.find(userId);
+    if (it != mAliasesMap.end())
+    {
+        std::string aliasB64 = it->second;
+        Buffer buf(aliasB64.size());
+        size_t decLen = base64urldecode(aliasB64.c_str(), aliasB64.size(), buf.buf(), buf.bufSize());
+        aliasBin.assign(buf.buf(), decLen);
```

Comment:
All the above for a simple conversion from B64 to binary seems too complicated for a simple operation. Please, pay attention to my adjustment.